google/cloud/aiplatform/experimental/vertex_model/serializers/dataloaders.py

```diff
@@ -15,35 +15,256 @@
+        dataloader_path (str): the path where the origin data used to construct the DataLoader
+                               resides.
+        obj (torch.utils.data.DataLoader): the pytorch DataLoader to serialize.
+        dataset_type (str): the intended use of the dataset (ie. training, testing)
 
+    Returns:
+        The GCS path pointing to the serialized DataLoader, the GCS path pointing to the
+        serialized origin data.
+    """
 
-def _deserialize_local_dataloader():
-    # read the data from user-designated staging bucket and
-    # reformat to a DataLoader
-    raise NotImplementedError
+    # TODO(b/195442091): Check if uri is actually a local path and write to a local
+    #                    location if that is the case.
 
+    # Create a client object
+    client = storage.Client(
+        project=initializer.global_config.project,
+        credentials=initializer.global_config.credentials,
+    )
 
-def _serialize_dataloader():
-    # introspect to determine which method is called
-    raise NotImplementedError
+    # Retrieve the source and blob names from the dataloader path
+    (
+        source_bucket_name,
+        source_blob_prefix,
+    ) = utils.extract_bucket_and_prefix_from_gcs_path(dataloader_path)
 
+    # Retrieve the source and blob names from the artifact URI
+    (
+        destination_bucket_name,
+        destination_blob_prefix,
+    ) = utils.extract_bucket_and_prefix_from_gcs_path(artifact_uri)
 
-def _deserialize_dataloader():
-    # introspect to determine which method is called
-    raise NotImplementedError
+    # Retrieve the blob and bucket name of the original GCS object
+    dataloader_path = pathlib.Path(dataloader_path)
+    local_file_name = dataset_type + "_" + dataloader_path.name
+    source_blob_name = local_file_name
+
+    if source_blob_prefix:
+        source_blob_name = "/".join([source_blob_prefix, source_blob_name])
+
+    source_bucket = client.bucket(source_bucket_name)
+    source_blob = source_bucket.blob(source_blob_name)
+
+    # Create a bucket and blob using the artifact URI
+    destination_bucket = client.bucket(destination_bucket_name)
+
+    destination_blob_name = source_blob_name
+    if destination_blob_prefix:
+        destination_blob_name = "/".join(
+            [destination_blob_prefix, destination_blob_name]
+        )
+
+    # Copy over the object from the source bucket to the new bucket
+    blob_copy = source_bucket.copy_blob(
+        source_blob, destination_bucket, destination_blob_name
+    )
+
+    data_gcs_path = "".join(
+        ["gs://", "/".join([destination_bucket_name, blob_copy.name])]
+    )
+
+    path = serialize_to_tmp_and_copy_to_gcs(
+        "my_" + dataset_type + "_dataloader.pth",
+        destination_bucket,
+        destination_blob_prefix,
+        torch.save,
+    )
+
+    return path, data_gcs_path
+
+
+def _serialize_local_dataloader(
+    artifact_uri: str,
+    dataloader_path: str,
```

Comment:
This should be `Optional` at `root` can be `None` based on the `_serialize_dataloader`. Also it looks like the `None` case isn't handled as `dataloader_path` is always passed to `pathlib.Path`